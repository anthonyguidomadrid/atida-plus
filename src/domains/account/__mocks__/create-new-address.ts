export const createNewAddress = {
  redirect: 'some-path',
  salutation: 'string',
  firstName: 'string',
  lastName: 'string',
  address1: 'string',
  address3: 'string',
  zipCode: 'string',
  city: 'string',
  country: 'string',
  iso2Code: 'string',
  phone: 'string',
  isDefaultShipping: true,
  isDefaultBilling: true
}

export const createNewAddressPayload = {
  data: {
    type: 'customers',
    attributes: {
      salutation: 'string',
      firstName: 'string',
      lastName: 'string',
      address1: 'string',
      address3: 'string',
      zipCode: 'string',
      city: 'string',
      country: 'string',
      iso2Code: 'string',
      phone: 'string',
      isDefaultShipping: true,
      isDefaultBilling: true
    }
  }
} as const

export const UpdatedJWTAndRefreshTokens = {
  refreshToken: 'newRefreshToken',
  JWT: 'newJWT',
  expiresIn: '12345'
}

export const sprykerUpdatedJWTAndRefreshTokens = {
  data: {
    type: '',
    id: '',
    attributes: {
      tokenType: 'Bearer',
      expiresIn: '12345',
      accessToken: 'newJWT',
      refreshToken: 'newRefreshToken',
      idCompanyUser: ''
    },
    links: {
      self: ''
    }
  },
  links: {
    self: ''
  }
}
