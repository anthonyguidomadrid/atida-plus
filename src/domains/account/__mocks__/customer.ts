export const createCustomer = {
  salutation: 'Mr',
  firstName: 'Test',
  lastName: 'McTest',
  email: 'some@email.com',
  password: 'somePassword1!',
  confirmPassword: 'somePassword1!',
  acceptedTerms: true,
  emailNotification: true,
  phone: '654654654',
  company: null,
  surcharge: false,
  addresses: {
    addresses: [
      {
        salutation: 'Mr',
        firstName: 'Test',
        lastName: 'McTest',
        address1: 'Brugstraat 123',
        zipCode: '1011 AD',
        city: 'Amsterdam',
        iso2Code: 'PT',
        phone: '088273791298432',
        isDefaultShipping: true,
        isDefaultBilling: true
      }
    ]
  }
}

export const sprykerCustomer = {
  data: {
    type: 'customers',
    id: 'some-id',
    attributes: {
      salutation: 'Mr',
      firstName: 'Test',
      lastName: 'McTest',
      email: 'some@email.com',
      password: 'somePassword1!',
      confirmPassword: 'somePassword1!',
      acceptedTerms: true,
      emailNotification: true,
      company: null,
      surcharge: false
    }
  }
} as const

export const sprykerCustomerWithAddresses = {
  data: {
    ...sprykerCustomer,
    attributes: {
      ...sprykerCustomer.data.attributes
    },
    relationships: {
      addresses: {
        data: [
          { type: 'addresses', id: 'address-id-1' },
          { type: 'addresses', id: 'address-id-2' }
        ]
      }
    }
  },
  included: [
    {
      type: 'addresses',
      id: 'address-id-1',
      attributes: {
        salutation: 'some-salutation',
        firstName: 'some-name',
        lastName: 'some-last-name',
        address1: 'some-address',
        address2: 'some-address2',
        address3: 'some-address3',
        zipCode: 'some-zip-code',
        city: 'some-city',
        country: 'some-country',
        iso2Code: 'some-iso-code',
        company: 'some-company',
        phone: 'some-phone',
        isDefaultShipping: true,
        isDefaultBilling: true,
        isTaxExempt: false
      }
    },
    {
      type: 'addresses',
      id: 'address-id-2',
      attributes: {
        salutation: 'some-other-salutation',
        firstName: 'some-other-name',
        lastName: 'some-other-last-name',
        address1: 'some-other-address',
        address2: 'some-other-address2',
        address3: 'some-other-address3',
        zipCode: 'some-other-zip-code',
        city: 'some-other-city',
        country: 'some-other-country',
        iso2Code: 'some-other-iso-code',
        company: 'some-other-company',
        phone: 'some-other-phone',
        isDefaultShipping: false,
        isDefaultBilling: false,
        isTaxExempt: false
      }
    }
  ]
}
