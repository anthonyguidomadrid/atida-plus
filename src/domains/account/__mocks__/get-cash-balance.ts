export const balance = {
  total: 20
}
