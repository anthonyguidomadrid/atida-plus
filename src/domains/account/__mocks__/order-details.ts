import { SHIPMENT_STATUSES } from '~config/constants/shipment-status'
import { OrderDetailsSingle } from '../types'

export const orderDetailsResult: OrderDetailsSingle = {
  id: 'PT--1234',
  type: 'orders',
  links: {
    self: 'http://self.order'
  },
  attributes: {
    createdAt: '2021-21-21 21:21:21',
    currencyIsoCode: 'EUR',
    priceMode: 'GROSS',
    status: {
      exported: {
        displayName: 'oms.state.exported',
        name: 'exported'
      }
    },
    productDiscounts: {
      voucher: {
        sumAmount: 3.99,
        voucherCode: 'discount',
        discountKey: 'discountKey'
      }
    },
    totals: {
      canceledTotal: 10,
      discountTotal: 10,
      expenseTotal: 10,
      grandTotal: 10,
      remunerationTotal: 10,
      surchargeTotal: 10,
      subtotal: 10,
      taxTotal: 10,
      shippingTotal: 10,
      itemTotal: 10
    },
    billingAddress: {
      address1: 'address 1',
      country: 'country',
      firstName: 'first name',
      iso2Code: 'GBGB',
      lastName: 'last name',
      isTaxExempt: false
    },
    shippingAddress: {
      address1: 'address 1',
      country: 'country',
      firstName: 'first name',
      iso2Code: 'GBGB',
      lastName: 'last name',
      isTaxExempt: false
    },
    items: [
      {
        metadata: {
          name: 'item',
          url: 'http://example.item/123456',
          pzn: '0000123456',
          brand: { code: 'Nice', label: 'Nice Brand' },
          format: { code: 'format1', label: 'Format 1' },
          productDatImage: 'http://example.item/image/thumbnail',
          thumbnailImage: 'http://example.item/image/thumbnail',
          mediumImage: 'http://example.item/image/medium',
          largeImage: 'http://example.item/image/large',
          productTileImage: 'http://example.item/image/productTile',
          productTileRetinaImage: 'http://example.item/image/produtTileRetina',
          description: 'description of item',
          shortDescription: 'short description of item',
          unitVolume: {
            unit: 'ml',
            amount: 10,
            unitLabel: 'ml'
          },
          price: {
            value: 10,
            currency: 'EUR'
          },
          pricePerUnit: {
            value: 10,
            currency: 'EUR',
            unit: 'ml'
          },
          rrp: {
            value: 10,
            currency: 'EUR'
          }
        },
        quantity: 10,
        sku: '0000000123456',
        sumGrossPrice: 10,
        unitGrossPrice: 10,
        calculatedDiscounts: [],
        canceledAmount: 0,
        isReturnable: false,
        name: 'item',
        productOptions: [{}],
        refundableAmount: 10,
        sumDiscountAmountAggregation: 10,
        sumDiscountAmountFullAggregation: 10,
        sumExpensePriceAggregation: 10,
        sumNetPrice: 10,
        sumPrice: 10,
        sumPriceToPayAggregation: 10,
        sumProductOptionPriceAggregation: 10,
        sumSubtotalAggregation: 10,
        sumTaxAmountFullAggregation: 10,
        taxRate: '1.00',
        taxRateAverageAggregation: '1.00',
        unitDiscountAmountAggregation: 10,
        unitDiscountAmountFullAggregation: 10,
        unitExpensePriceAggregation: 10,
        unitNetPrice: 10,
        unitPrice: 10,
        unitPriceToPayAggregation: 10,
        unitProductOptionPriceAggregation: 10,
        unitSubtotalAggregation: 10,
        unitTaxAmountFullAggregation: 10,
        uuid: 'akdsbfakshdbf-askdfbaksd-aksdjbfaks-asjdhfvajsdhf',
        id: '755632207021',
        isPromo: false
      }
    ],
    calculatedDiscounts: {},
    expenses: [
      {
        idSalesExpense: 10,
        name: 'expense',
        sumGrossPrice: 10,
        sumNetPrice: 10,
        sumPrice: 10,
        sumPriceToPayAggregation: 10,
        sumTaxAmount: 10,
        type: 'expense_type',
        unitGrossPrice: 10,
        unitNetPrice: 10,
        unitPriceToPayAggregation: 10,
        unitTaxAmount: 10
      }
    ],
    payments: [
      {
        amount: 10,
        paymentMethod: 'paymentMethod',
        paymentProvider: 'paymentProvider',
        paymentReference: 'PMT-PT--1234'
      }
    ],
    shipments: [
      {
        carrierName: 'carrier',
        currencyIsoCode: 'EUR',
        defaultGrossPrice: 10,
        defaultNetPrice: 10,
        shipmentMethodName: 'express_shipping',
        maxDeliveryDays: 3,
        minDeliveryDays: 1,
        erpShipmentStatus: 'in_progres' as typeof SHIPMENT_STATUSES[number],
        erpTrackingUrl: 'some-tracking-url',
        erpTrackingReference: 'some-tracking-url'
      }
    ]
  }
}

export const orderDetailsResultWithoutItems: OrderDetailsSingle = {
  id: 'PT--1234',
  type: 'orders',
  links: {
    self: 'http://self.order'
  },
  attributes: {
    createdAt: '2021-21-21 21:21:21',
    currencyIsoCode: 'EUR',
    priceMode: 'GROSS',
    status: {
      exported: {
        displayName: 'oms.state.exported',
        name: 'exported'
      }
    },
    productDiscounts: {
      voucher: {
        sumAmount: 3.99,
        voucherCode: 'discount',
        discountKey: 'discountKey'
      }
    },
    totals: {
      canceledTotal: 10,
      discountTotal: 10,
      expenseTotal: 10,
      grandTotal: 10,
      remunerationTotal: 10,
      surchargeTotal: 10,
      subtotal: 10,
      taxTotal: 10,
      shippingTotal: 10,
      itemTotal: 10
    },
    billingAddress: {
      address1: 'address 1',
      country: 'country',
      firstName: 'first name',
      iso2Code: 'GBGB',
      lastName: 'last name',
      isTaxExempt: false
    },
    shippingAddress: {
      address1: 'address 1',
      country: 'country',
      firstName: 'first name',
      iso2Code: 'GBGB',
      lastName: 'last name',
      isTaxExempt: false
    },
    items: [],
    calculatedDiscounts: {},
    expenses: [
      {
        idSalesExpense: 10,
        name: 'expense',
        sumGrossPrice: 10,
        sumNetPrice: 10,
        sumPrice: 10,
        sumPriceToPayAggregation: 10,
        sumTaxAmount: 10,
        type: 'expense_type',
        unitGrossPrice: 10,
        unitNetPrice: 10,
        unitPriceToPayAggregation: 10,
        unitTaxAmount: 10
      }
    ],
    payments: [
      {
        amount: 10,
        paymentMethod: 'paymentMethod',
        paymentProvider: 'paymentProvider',
        paymentReference: 'PMT-PT--1234'
      }
    ],
    shipments: [
      {
        carrierName: 'carrier',
        currencyIsoCode: 'EUR',
        defaultGrossPrice: 10,
        defaultNetPrice: 10,
        shipmentMethodName: 'express_shipping',
        maxDeliveryDays: 3,
        minDeliveryDays: 1,
        erpShipmentStatus: 'in_progres' as typeof SHIPMENT_STATUSES[number],
        erpTrackingUrl: 'some-tracking-url',
        erpTrackingReference: 'some-tracking-url'
      }
    ]
  }
}
