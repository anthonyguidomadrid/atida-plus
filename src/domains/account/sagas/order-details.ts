import { SagaIterator } from '@redux-saga/types'
import type { PayloadAction } from '@reduxjs/toolkit'
import type { AxiosResponse } from 'axios'
import { call, getContext, put, takeLatest, delay } from 'redux-saga/effects'
import { createClient, logger } from '~helpers'
import {
  orderDetailsFailure,
  orderDetailsFulfill,
  orderDetailsRequest,
  orderDetailsSuccess,
  orderDetailsTrigger
} from '../slices'
import type { OrderDetailsPayload, OrderDetailsTriggerPayload } from '../types'
import { getErrorMessage } from '~helpers/error'

function* doOrderDetailsSaga({
  payload
}: PayloadAction<OrderDetailsTriggerPayload>) {
  try {
    yield put(orderDetailsRequest())

    const locale: string = yield getContext('locale')
    const client = createClient({ locale, addAnonymousCustomerUniqueId: false })

    const apiResponse: AxiosResponse<OrderDetailsPayload> = yield call(
      client.get,
      '/api/account/order-details',
      { params: payload }
    )

    yield put(orderDetailsSuccess(apiResponse.data))

    return apiResponse
  } catch (error) {
    const errorMessages = getErrorMessage(error)
    logger.error(errorMessages)
    if (Array.isArray(errorMessages)) {
      yield put(orderDetailsFailure({ message: errorMessages[0] }))
    } else {
      yield put(
        orderDetailsFailure({
          message: errorMessages
        })
      )
    }
  } finally {
    yield delay(0)
    yield put(orderDetailsFulfill())
  }
}

export function* orderDetailsSaga(): SagaIterator {
  yield takeLatest(orderDetailsTrigger, doOrderDetailsSaga)
}
