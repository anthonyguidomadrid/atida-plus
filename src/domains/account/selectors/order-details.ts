import { createSelector } from '@reduxjs/toolkit'
import { RootState } from '~domains/redux'

const selectorOrderDetails = (state: RootState) =>
  state?.client?.account?.['order-details']

export const selectOrderDetailsIsLoading = createSelector(
  selectorOrderDetails,
  orderHistory => orderHistory?.isLoading ?? false
)

export const selectOrderDetails = createSelector(
  selectorOrderDetails,
  orderDetails => orderDetails?.orderDetails
)

export const selectOrderDetailsWasSuccess = createSelector(
  selectorOrderDetails,
  orderDetails => orderDetails?.wasSuccess
)

export const selectOrderDetailsWasError = createSelector(
  selectorOrderDetails,
  orderDetails => orderDetails?.wasError
)

export const selectOrderDetailsItems = createSelector(
  selectorOrderDetails,
  orderDetails => orderDetails?.orderDetails?.attributes?.items
)

export const selectOrderDetailsCurrency = createSelector(
  selectorOrderDetails,
  orderDetails => orderDetails?.orderDetails?.attributes?.currencyIsoCode
)

export const selectOrderDetailsDiscountTotal = createSelector(
  selectorOrderDetails,
  orderDetails => orderDetails?.orderDetails?.attributes?.productDiscounts
)

export const selectOrderDiscounts = createSelector(
  selectorOrderDetails,
  orderDetails => orderDetails?.orderDetails?.attributes?.calculatedDiscounts
)

export const selectOrderDetailsTotals = createSelector(
  selectOrderDetails,
  data => ({
    subTotal: data?.attributes?.totals?.subtotal ?? 0,
    shippingTotal: data?.attributes?.totals?.shippingTotal ?? 0,
    grandTotal: data?.attributes?.totals?.grandTotal ?? 0,
    surchargeTotal: data?.attributes?.totals?.surchargeTotal ?? 0,
    rrpDiscountTotal: data?.attributes?.totals?.rrpDiscountTotal ?? 0,
    rewardTotal: data?.attributes?.totals?.rewardTotal ?? 0,
    loyaltySpent: data?.attributes?.totals?.loyaltySpent ?? 0,
    amountPaidByRealPayment:
      data?.attributes?.totals?.amountPaidByRealPayment ?? 0,
    itemTotal: data?.attributes?.totals?.itemTotal ?? 0
  })
)

const selectOrderDetailsItemQuantities = createSelector(
  selectOrderDetailsItems,
  items => items?.map(item => item.quantity)
)

export const selectOrderDetailsNumberOfItems = createSelector(
  selectOrderDetailsItemQuantities,
  quantities =>
    quantities?.reduce((total = 0, quantity = 0) => total + quantity, 0) ?? 0
)

export const selectOrderDetailsCoupons = createSelector(
  selectorOrderDetails,
  orderDetails => orderDetails?.orderDetails?.attributes?.productDiscounts
)

export const selectOrderDetailsIsTaxExempt = createSelector(
  selectorOrderDetails,
  orderDetails =>
    orderDetails?.orderDetails?.attributes?.shippingAddress?.isTaxExempt
)

export const selectOrderDetailsFirstName = createSelector(
  selectorOrderDetails,
  orderDetails =>
    orderDetails?.orderDetails?.attributes?.shippingAddress?.firstName
)

export const selectOrderDetailsShippingTotal = createSelector(
  selectorOrderDetails,
  orderDetails =>
    orderDetails?.orderDetails?.attributes?.expenses[0]
      ?.sumPriceToPayAggregation
)

export const selectDeliveryDays = createSelector(
  selectorOrderDetails,
  orderDetails => ({
    minDeliveryDays:
      orderDetails?.orderDetails?.attributes?.shipments[0]?.minDeliveryDays,
    maxDeliveryDays:
      orderDetails?.orderDetails?.attributes?.shipments[0]?.maxDeliveryDays
  })
)

export const selectErpShipping = createSelector(
  selectorOrderDetails,
  orderDetails => ({
    erpShipmentStatus:
      orderDetails?.orderDetails?.attributes?.shipments[0].erpShipmentStatus,
    erpTrackingUrl:
      orderDetails?.orderDetails?.attributes?.shipments[0]?.erpTrackingUrl,
    erpTrackingReference:
      orderDetails?.orderDetails?.attributes?.shipments[0]?.erpTrackingReference
  })
)
