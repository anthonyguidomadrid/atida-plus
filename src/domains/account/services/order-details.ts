import getConfig from 'next/config'
import { createClient } from '~helpers'
import type {
  OrderDetailsParams,
  OrderDetailsPayload,
  OrderDetailsSingle
} from '../types'
import {
  createElasticSearchClient,
  getElasticSearchProductIndex
} from '~domains/elasticsearch/helpers/client'
import {
  ElasticSearchProduct,
  normalizeElasticSearchProduct
} from '~domains/product'

if (typeof window !== 'undefined') {
  throw new Error('This file should not be called client-side')
}

export const orderDetails = async ({
  locale,
  accessToken,
  orderId
}: OrderDetailsParams): Promise<OrderDetailsPayload> => {
  const { serverRuntimeConfig } = getConfig()

  const client = createClient({
    locale,
    options: { baseURL: serverRuntimeConfig.ecommerceHost[locale] }
  })

  const { data: order } = await client.get<{ data: OrderDetailsSingle }>(
    `/orders/${orderId}`,
    {
      headers: {
        Authorization: `Bearer ${accessToken}`
      }
    }
  )

  if (order.data.attributes.items.length > 0) {
    const elasticClient = createElasticSearchClient()
    const elasticResponse = await elasticClient.mget<{
      docs: ElasticSearchProduct[]
    }>({
      index: getElasticSearchProductIndex(locale),
      body: {
        ids: Array.from(
          new Set(order?.data?.attributes?.items?.map(({ sku }) => sku))
        )
      }
    })
    const elasticData = order?.data?.attributes?.items?.map(basketItem => {
      const isPromo = basketItem?.isPromo
      const isFullyDiscounted = basketItem?.isFullyDiscounted
      basketItem.id = isPromo ? `${basketItem.sku}-promotion` : basketItem.sku
      const doc = elasticResponse?.body?.docs?.find(
        doc => doc._id === basketItem.sku
      )
      doc && (doc.id = basketItem.id)

      return normalizeElasticSearchProduct(
        locale,
        doc,
        undefined,
        undefined,
        isPromo,
        isFullyDiscounted
      )
    })

    const orderDetailsData = order?.data
    const orderDetailsAttributes = orderDetailsData?.attributes
    const items = orderDetailsAttributes?.items

    const orderItems = items?.map(item => {
      if (!elasticData) return item

      const { id: elasticDataItemId, ...elasticDataItem } =
        elasticData.find(eld => eld.id === item.id) ?? {}

      if (!elasticDataItem) return item

      return {
        ...item,
        metadata: elasticDataItem
      }
    })

    return {
      ...orderDetailsData,
      attributes: {
        ...orderDetailsAttributes,
        items: orderItems
      }
    }
  }

  const orderDetailsData = order?.data
  const orderDetailsAttributes = orderDetailsData?.attributes

  return {
    ...orderDetailsData,
    attributes: {
      ...orderDetailsAttributes,
      items: []
    }
  }
}
