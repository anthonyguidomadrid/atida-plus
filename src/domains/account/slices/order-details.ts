import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'
import type {
  OrderDetailsPayload,
  OrderDetailsSingle,
  OrderDetailsTriggerPayload
} from '../types'

type OrderDetailsState = {
  isLoading: boolean
  wasSuccess: boolean
  wasError: boolean
  error?: string
  orderDetails?: OrderDetailsSingle
}

const initialState: OrderDetailsState = {
  isLoading: false,
  wasSuccess: false,
  wasError: false
}

export const orderDetailsSlice = createSlice({
  name: 'account/order-details',
  initialState,
  reducers: {
    trigger(_state, _action: PayloadAction<OrderDetailsTriggerPayload>) {
      // no-op, used for triggering saga
    },
    request(state) {
      state.isLoading = true
      state.wasSuccess = false
      state.wasError = false

      delete state.error
    },
    success(state, action: PayloadAction<OrderDetailsPayload>) {
      const data = action.payload

      state.wasSuccess = true
      state.orderDetails = data
    },
    failure(state, action: PayloadAction<{ message: string }>) {
      const { message } = action.payload

      state.wasError = true
      state.error = message
    },
    fulfill(state) {
      state.isLoading = false
    },
    resetState(state) {
      state.isLoading = false
      state.wasSuccess = false
      state.wasError = false
      delete state.error
      delete state.orderDetails
    }
  }
})
