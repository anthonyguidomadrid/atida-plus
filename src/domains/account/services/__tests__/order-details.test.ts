/**
 * @jest-environment node
 */
import axios from 'axios'
import { createClient } from '~helpers'
import { orderDetails } from '../order-details'
import {
  orderDetailsResult,
  orderDetailsResultWithoutItems
} from '../../__mocks__/order-details'
// @ts-ignore
import { mget } from '@elastic/elasticsearch'
import { elasticsearchProducts } from '~domains/basket/__mocks__/basket'

describe(orderDetails, () => {
  describe('creates the client and passes the locale to the spryker request', () => {
    beforeEach(() => {
      ;(axios.get as jest.Mock).mockResolvedValueOnce({
        data: {
          data: orderDetailsResult
        }
      })
    })

    it('gets details for order PT--1234', async () => {
      await orderDetails({
        locale: 'en-gb',
        accessToken: 'some-token',
        orderId: 'PT--1234'
      })

      expect(createClient).toHaveBeenCalledWith({
        locale: 'en-gb',
        options: {
          baseURL: 'spryker.co.uk'
        }
      })

      expect(axios.get).toHaveBeenCalledWith('/orders/PT--1234', {
        headers: {
          Authorization: 'Bearer some-token'
        }
      })
    })
  })
  describe('test order details ', () => {
    it('with populated data', async () => {
      mget.mockResolvedValue(elasticsearchProducts)
      ;(axios.get as jest.Mock).mockResolvedValue({
        data: {
          data: orderDetailsResult,
          links: []
        }
      })
      const res = await orderDetails({
        locale: 'en-gb',
        accessToken: 'some-token',
        orderId: 'PT--1234'
      })
      if (res) {
        expect(res).toMatchSnapshot()
      }
    })

    it('without data items', async () => {
      mget.mockResolvedValue(elasticsearchProducts)
      ;(axios.get as jest.Mock).mockResolvedValue({
        data: {
          data: orderDetailsResultWithoutItems,
          links: []
        }
      })
      const res = await orderDetails({
        locale: 'en-gb',
        accessToken: 'some-token',
        orderId: 'PT--1234'
      })
      if (res) {
        expect(res).toMatchSnapshot()
      }
    })
  })
})
