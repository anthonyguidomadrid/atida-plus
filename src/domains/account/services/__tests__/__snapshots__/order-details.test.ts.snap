// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderDetails test order details  with populated data 1`] = `
Object {
  "attributes": Object {
    "billingAddress": Object {
      "address1": "address 1",
      "country": "country",
      "firstName": "first name",
      "isTaxExempt": false,
      "iso2Code": "GBGB",
      "lastName": "last name",
    },
    "calculatedDiscounts": Object {},
    "createdAt": "2021-21-21 21:21:21",
    "currencyIsoCode": "EUR",
    "expenses": Array [
      Object {
        "idSalesExpense": 10,
        "name": "expense",
        "sumGrossPrice": 10,
        "sumNetPrice": 10,
        "sumPrice": 10,
        "sumPriceToPayAggregation": 10,
        "sumTaxAmount": 10,
        "type": "expense_type",
        "unitGrossPrice": 10,
        "unitNetPrice": 10,
        "unitPriceToPayAggregation": 10,
        "unitTaxAmount": 10,
      },
    ],
    "items": Array [
      Object {
        "calculatedDiscounts": Array [],
        "canceledAmount": 0,
        "id": "0000000123456",
        "isPromo": false,
        "isReturnable": false,
        "metadata": Object {},
        "name": "item",
        "productOptions": Array [
          Object {},
        ],
        "quantity": 10,
        "refundableAmount": 10,
        "sku": "0000000123456",
        "sumDiscountAmountAggregation": 10,
        "sumDiscountAmountFullAggregation": 10,
        "sumExpensePriceAggregation": 10,
        "sumGrossPrice": 10,
        "sumNetPrice": 10,
        "sumPrice": 10,
        "sumPriceToPayAggregation": 10,
        "sumProductOptionPriceAggregation": 10,
        "sumSubtotalAggregation": 10,
        "sumTaxAmountFullAggregation": 10,
        "taxRate": "1.00",
        "taxRateAverageAggregation": "1.00",
        "unitDiscountAmountAggregation": 10,
        "unitDiscountAmountFullAggregation": 10,
        "unitExpensePriceAggregation": 10,
        "unitGrossPrice": 10,
        "unitNetPrice": 10,
        "unitPrice": 10,
        "unitPriceToPayAggregation": 10,
        "unitProductOptionPriceAggregation": 10,
        "unitSubtotalAggregation": 10,
        "unitTaxAmountFullAggregation": 10,
        "uuid": "akdsbfakshdbf-askdfbaksd-aksdjbfaks-asjdhfvajsdhf",
      },
    ],
    "payments": Array [
      Object {
        "amount": 10,
        "paymentMethod": "paymentMethod",
        "paymentProvider": "paymentProvider",
        "paymentReference": "PMT-PT--1234",
      },
    ],
    "priceMode": "GROSS",
    "productDiscounts": Object {
      "voucher": Object {
        "discountKey": "discountKey",
        "sumAmount": 3.99,
        "voucherCode": "discount",
      },
    },
    "shipments": Array [
      Object {
        "carrierName": "carrier",
        "currencyIsoCode": "EUR",
        "defaultGrossPrice": 10,
        "defaultNetPrice": 10,
        "erpShipmentStatus": "in_progres",
        "erpTrackingReference": "some-tracking-url",
        "erpTrackingUrl": "some-tracking-url",
        "maxDeliveryDays": 3,
        "minDeliveryDays": 1,
        "shipmentMethodName": "express_shipping",
      },
    ],
    "shippingAddress": Object {
      "address1": "address 1",
      "country": "country",
      "firstName": "first name",
      "isTaxExempt": false,
      "iso2Code": "GBGB",
      "lastName": "last name",
    },
    "status": Object {
      "exported": Object {
        "displayName": "oms.state.exported",
        "name": "exported",
      },
    },
    "totals": Object {
      "canceledTotal": 10,
      "discountTotal": 10,
      "expenseTotal": 10,
      "grandTotal": 10,
      "itemTotal": 10,
      "remunerationTotal": 10,
      "shippingTotal": 10,
      "subtotal": 10,
      "surchargeTotal": 10,
      "taxTotal": 10,
    },
  },
  "id": "PT--1234",
  "links": Object {
    "self": "http://self.order",
  },
  "type": "orders",
}
`;

exports[`orderDetails test order details  without data items 1`] = `
Object {
  "attributes": Object {
    "billingAddress": Object {
      "address1": "address 1",
      "country": "country",
      "firstName": "first name",
      "isTaxExempt": false,
      "iso2Code": "GBGB",
      "lastName": "last name",
    },
    "calculatedDiscounts": Object {},
    "createdAt": "2021-21-21 21:21:21",
    "currencyIsoCode": "EUR",
    "expenses": Array [
      Object {
        "idSalesExpense": 10,
        "name": "expense",
        "sumGrossPrice": 10,
        "sumNetPrice": 10,
        "sumPrice": 10,
        "sumPriceToPayAggregation": 10,
        "sumTaxAmount": 10,
        "type": "expense_type",
        "unitGrossPrice": 10,
        "unitNetPrice": 10,
        "unitPriceToPayAggregation": 10,
        "unitTaxAmount": 10,
      },
    ],
    "items": Array [],
    "payments": Array [
      Object {
        "amount": 10,
        "paymentMethod": "paymentMethod",
        "paymentProvider": "paymentProvider",
        "paymentReference": "PMT-PT--1234",
      },
    ],
    "priceMode": "GROSS",
    "productDiscounts": Object {
      "voucher": Object {
        "discountKey": "discountKey",
        "sumAmount": 3.99,
        "voucherCode": "discount",
      },
    },
    "shipments": Array [
      Object {
        "carrierName": "carrier",
        "currencyIsoCode": "EUR",
        "defaultGrossPrice": 10,
        "defaultNetPrice": 10,
        "erpShipmentStatus": "in_progres",
        "erpTrackingReference": "some-tracking-url",
        "erpTrackingUrl": "some-tracking-url",
        "maxDeliveryDays": 3,
        "minDeliveryDays": 1,
        "shipmentMethodName": "express_shipping",
      },
    ],
    "shippingAddress": Object {
      "address1": "address 1",
      "country": "country",
      "firstName": "first name",
      "isTaxExempt": false,
      "iso2Code": "GBGB",
      "lastName": "last name",
    },
    "status": Object {
      "exported": Object {
        "displayName": "oms.state.exported",
        "name": "exported",
      },
    },
    "totals": Object {
      "canceledTotal": 10,
      "discountTotal": 10,
      "expenseTotal": 10,
      "grandTotal": 10,
      "itemTotal": 10,
      "remunerationTotal": 10,
      "shippingTotal": 10,
      "subtotal": 10,
      "surchargeTotal": 10,
      "taxTotal": 10,
    },
  },
  "id": "PT--1234",
  "links": Object {
    "self": "http://self.order",
  },
  "type": "orders",
}
`;
